hls4ml/templates/vitis/nnet_utils/nnet_conv1d_latency.h

```diff
@@ -85,5 +85,85 @@ void conv_1d_latency_cl(data_T data[CONFIG_T::in_width * CONFIG_T::n_chan],
     }
 }
 
+template <class data_T, class res_T, typename CONFIG_T>
+void pointwise_conv_1d_latency_cl(data_T data[CONFIG_T::in_width * CONFIG_T::n_chan / CONFIG_T::reuse_factor],
+                                  res_T res[CONFIG_T::out_width * CONFIG_T::n_filt / CONFIG_T::reuse_factor],
+                                  typename CONFIG_T::weight_t weights[CONFIG_T::n_chan * CONFIG_T::n_filt],
+                                  typename CONFIG_T::bias_t biases[CONFIG_T::n_filt]) {
+    assert(CONFIG_T::filt_width == 1);
+
+    typename CONFIG_T::accum_t mult[CONFIG_T::out_width * CONFIG_T::n_filt * CONFIG_T::n_chan / CONFIG_T::reuse_factor];
+    typename CONFIG_T::accum_t acc[CONFIG_T::out_width / CONFIG_T::reuse_factor][CONFIG_T::n_filt];
+
+    #pragma HLS ARRAY_PARTITION variable=mult complete dim=0
+    #pragma HLS ARRAY_PARTITION variable=acc complete dim=0
+
+    // Use a function_instantiate in case it helps to explicitly optimize unchanging weights/biases
+    #pragma HLS function_instantiate variable=weights,biases
+
+    // Parallel mode
+    #pragma HLS PIPELINE II=CONFIG_T::reuse_factor
+    #pragma HLS ARRAY_PARTITION variable=weights complete dim=0
+    #pragma HLS ARRAY_PARTITION variable=biases complete dim=0
+
+    // Limit multipliers to control parallelization
+    constexpr unsigned multiplier_limit = DIV_ROUNDUP(
```

Comment:
This was problematic before, and we moved to setting `multiplier_limit` in python and using it here. For consistency, we should not re-introduce old approaches.

Reply:
updated to use the `multiplier_limit` from the config here: https://github.com/jmduarte/hls4ml/pull/22/commits/d56dc7349e94ad8b6aaefe5b0f22385d3d9a9e03, but need to check the value is the same

Reply:
ok, fixed the multiplier limit now... but it would be good to warn the user somehow if they use a `reuse_factor` that doesn't divide `in_width`

related: it would probably be beneficial to factorize the two into a `reuse_factor` (that limits the multipliers) and a `parallelization_factor` (that controls how many times the conv is split into separate calls)